src/lightning/data/streaming/dataset.py

```diff
@@ -68,12 +73,24 @@ def __init__(
 
         self.item_loader = item_loader
         self.shuffle: bool = shuffle
-        self.drop_last = drop_last
+        self.distributed_env = _DistributedEnv.detect()
+
+        if self.distributed_env.world_size > 1:
+            if drop_last is None or drop_last is True:
+                self.drop_last = True
+            else:
+                logger.warn(
+                    "You are running in a distributed environment and drop_last needs to be False. "
+                    "This can cause your training to hang if you are relying on distributed collectives."
+                )
```

Comment:
Language here is a bit strange - it says "drop_last needs to be False" but drop last is already False here? Also it then just carries on anyway so maybe it should instead override it and make drop_last true, or just throw an exception

Reply:
Improve the messaging. It shouldn't raise an exception as this is correct for the validation set. However, it is important for users to be aware of potential issues.